Task BSJ2-68351

EkvinoksLeyn opened the pull request:
1. Добавил класс User
2. Объявил инициализацию констант VALID_JOBS, VALID_ADRESSES
3. Добавил конструктор в классе User

EkvinoksLeyn committed:
Task BSJ2-68351

1. Добавил класс User
2. Объявил инициализацию констант VALID_JOBS, VALID_ADRESSES
3. Добавил конструктор в классе User

TheWandererr commented on src/main/java/school/faang/Main.java:
```diff
@@ -1,4 +1,14 @@
 
 public class Main {
+    public static void main(String[] args) {
+        try {
```
Так себе подход. Все оборачивать в try catch дорого и неэффективно. Лучше иметь отдельный метод на создание пользователя

TheWandererr commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,63 @@
+    private final String name;
+    private final int age;
+    private final String job;
+    private final String address;
```
Для текущей задачи пойдет. Но в теории адрес человека может меняться:)

TheWandererr commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,63 @@
+    private final String address;
+
+    public static final Set<String> VALID_JOBS = new HashSet<>();
+    public static final Set<String> VALID_ADDRESSES = new HashSet<>();
```
Стоит ли иметь public? Как будто бы используется только внутри текущего класса

TheWandererr commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,63 @@
+            throw new IllegalArgumentException("Имя не может быть пустым");
+        }
+
+        if (age < 18) {
```
magic number. А как насчет отрицательного возраста?

TheWandererr commented on src/main/java/school/faang/User.java:
```diff
@@ -0,0 +1,63 @@
+        this.address = address;
+    }
+
+    public String getName() {
```
У ломбока есть @getter. А нужны ли эти методы в целом?

EkvinoksLeyn committed:
Task BSJ2-68351

1. Добавил класс User
2. Объявил инициализацию констант VALID_JOBS, VALID_ADRESSES
3. Добавил конструктор в классе User

EkvinoksLeyn committed:
Task BSJ2-68351

1. Добавил класс User
2. Объявил инициализацию констант VALID_JOBS, VALID_ADRESSES
3. Добавил конструктор в классе User

EkvinoksLeyn committed:
Task BSJ2-68351

1. Добавил класс User
2. Объявил инициализацию констант VALID_JOBS, VALID_ADRESSES
3. Добавил конструктор в классе User

EkvinoksLeyn committed:
Merge remote-tracking branch 'origin/Ekvi_BJS2-68351' into Ekvi_BJS2-68351_V2
